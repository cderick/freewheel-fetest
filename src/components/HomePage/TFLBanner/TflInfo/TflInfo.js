import React from 'react'
import PropTypes from 'prop-types'
import s from './TflInfo.module.scss'

class TflInfo extends React.Component {
    constructor(props) {
		super(props)
		this.sortData = this.sortData.bind(this)
    }
    
    sortData(allData){
        return allData.sort((a, b) => {
            if (a.name > b.name) return 1
            if (a.name < b.name) return -1
            if (a.modeName > b.modeName) return 1
            if (a.modeName < b.modeName) return -1
            return 0
        })
    }

	render() {
		if (!this.props.tflInfo) return null
        const sortedData = this.sortData(this.props.tflInfo)
		return (
            <div className="col-5 m-5">
                {sortedData && sortedData.length
                    && sortedData.map((transport, ind) => (
                        <section key={`tlfInfo${ind}`} className={s.dataClick} onClick={() => this.props.handleClick(transport)}>
                            <p>
                                <strong>{transport.modeName}: </strong>{`${transport.name}`}
                                <span>{(transport.serviceTypes.filter(workTime => (workTime.name).toLowerCase() === 'night')).length > 0 && <i className="fas fa-moon"></i>}</span>
                                <span>{(transport.lineStatuses.filter(status => status.statusSeverity !== 10)).length > 0 
                                && <i className="fas fa-ban"></i>}</span>
                            </p>
                        </section>
                    ))}
            </div>
		)
	}
}

TflInfo.propTypes = {
	handleClick: PropTypes.func,
	tflInfo: PropTypes.arrayOf(PropTypes.any),
}

TflInfo.defaultProps = {
	tflInfo: null,
}

export default TflInfo
