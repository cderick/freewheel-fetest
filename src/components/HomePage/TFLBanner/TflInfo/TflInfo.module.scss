.dataClick {
  cursor: pointer;
  background-color: #f2dfff;
  border-radius: 5px;
  padding: 10px;
  margin: 5px;
  color: #070431;

  p {
    padding: 0;
    margin: 0;

    span {
      margin-left: 10px;
    }
  }
}
