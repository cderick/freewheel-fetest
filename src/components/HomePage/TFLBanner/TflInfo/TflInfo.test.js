import React from 'react'
import renderer from 'react-test-renderer'
import TflInfo from './TflInfo'
import mockData from './__mock__/mock'

const componentNoprops = renderer.create(<TflInfo />)
const componentWithProps = renderer.create(<TflInfo {...mockData}  />)

describe('Test on <TflInfo /> component', () => {
   test('should not render with no props', () => {
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
   test('should render with all props', () => {
      const treeWithprops = componentWithProps.toJSON()
      expect(treeWithprops).toMatchSnapshot()
   })
})