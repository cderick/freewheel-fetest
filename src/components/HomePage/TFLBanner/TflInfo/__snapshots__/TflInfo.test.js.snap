// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <TflInfo /> component should not render with no props 1`] = `null`;

exports[`Test on <TflInfo /> component should render with all props 1`] = `
<div
  className="col-5 m-5"
>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Bakerloo
      <span />
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Central
      <span>
        <i
          className="fas fa-moon"
        />
      </span>
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Circle
      <span />
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        dlr
        : 
      </strong>
      DLR
      <span />
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      District
      <span />
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Hammersmith & City
      <span />
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Jubilee
      <span>
        <i
          className="fas fa-moon"
        />
      </span>
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        overground
        : 
      </strong>
      London Overground
      <span>
        <i
          className="fas fa-moon"
        />
      </span>
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Metropolitan
      <span />
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Northern
      <span>
        <i
          className="fas fa-moon"
        />
      </span>
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Piccadilly
      <span>
        <i
          className="fas fa-moon"
        />
      </span>
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Victoria
      <span>
        <i
          className="fas fa-moon"
        />
      </span>
      <span />
    </p>
  </section>
  <section
    className="dataClick"
    onClick={[Function]}
  >
    <p>
      <strong>
        tube
        : 
      </strong>
      Waterloo & City
      <span />
      <span />
    </p>
  </section>
</div>
`;
